### PR TITLE
Add Query Monitor support

### PR DESCRIPTION
**Pull request by joehoyle**
Working on X-Ray locally can be quite tough, as it requires either hooking it up to a live AWS account, or manually printing out data that would be sent to the daemon. This adds a new Query Monitor panel for all the segments sent to AWS X-Ray for inspection. It also adds the Flamegraph for the request in a new panel, so it can be used as a profiling tool too.

Screenshots:

### CODE REVIEW

**Comment by joehoyle** [comment]
- [ ] Need to only enqueue scripts when Query Monitor is actually enabled

**Comment by rmccue** [comment]
Does this affect page load times? I suspect this is a lot of data.

**Comment by joehoyle** [comment]
@rmccue not really, this is all data that is already processed, so it's just a case of outputting it in the Outputter, so more page weight, though I don't think as much as say, the rest of the query monitor output.

File: inc/query_monitor/class-output-flamegraph-html.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+		$this->after_non_tabular_output();
+	}
+
+	public function panel_menu( array $menu ) : array {
+		$menu['aws-xray']['children'][] = [
+			'title' => 'Flamegraph',
+			'id'    => 'query-monitor-aws-xray-flamegraph',
+			'href'  => '#qm-aws-xray-flamegraph',
+
```
Comment (roborourke) [inline inc/query_monitor/class-output-flamegraph-html.php]: Bonus newline
```suggestion
```

File: inc/query_monitor/class-output-flamegraph-html.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		}
+
+		if ( ! $xhprof ) {
+			return;
+		}
+
+		?>
+		<?php $this->before_non_tabular_output( 'qm-aws-xray-flamegraph' ); ?>
+		<caption>
+			<h2>Sampled Profile (5ms intervals)</h2>
```
Comment (roborourke) [inline inc/query_monitor/class-output-flamegraph-html.php]: Use `esc_html__()` wrapper here.

File: inc/query_monitor/class-output-flamegraph-html.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+		</caption>
+		<div class="aws-xray-flamegraph"><?php echo wp_json_encode( $xhprof ) ?></div>
+		<?php
+
+		$this->after_non_tabular_output();
+	}
+
+	public function panel_menu( array $menu ) : array {
+		$menu['aws-xray']['children'][] = [
+			'title' => 'Flamegraph',
```
Comment (roborourke) [inline inc/query_monitor/class-output-flamegraph-html.php]: This is sort of a noun but could be up to translators to decide.

File: inc/query_monitor/class-output-html.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	public function __construct( QM_Collector $collector ) {
+		parent::__construct( $collector );
+		add_filter( 'qm/output/panel_menus', [ $this, 'panel_menu' ], 40 );
+	}
+
+	public function output() {
+		?>
+		<?php $this->before_tabular_output(); ?>
+		<caption>
+			<h2>Trace ID: <?php echo esc_html( get_root_trace_id() ) ?></h2>
```
Comment (roborourke) [inline inc/query_monitor/class-output-html.php]: Translation function for `Trace ID`

File: inc/query_monitor/class-output-html.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		</caption>
+		<thead>
+			<tr>
+				<th>Segment Name</th>
+				<th>In Progress</th>
+				<th>Time</th>
+				<th>Segment</th>
```
Comment (roborourke) [inline inc/query_monitor/class-output-html.php]: Make these translatable

**Comment by joehoyle** [comment]
Thanks rob, will get on this!

**Comment by joehoyle** [comment]
Cheers @roborourke, I added translation functions and also some more doc blocks.

File: inc/query_monitor/class-output-flamegraph-html.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+		}
+
+		if ( ! $xhprof ) {
+			return;
+		}
+
+		?>
+		<?php $this->before_non_tabular_output( 'qm-aws-xray-flamegraph' ); ?>
+		<caption>
+			<?php /* translators: Number of miliseconds/ */ ?>
```
Comment (roborourke) [inline inc/query_monitor/class-output-flamegraph-html.php]:
```suggestion
			<?php /* translators: %d = Number of milliseconds */ ?>
```

**Pull request merged**